### ISSUE TITLE
I need to move the canvas with the right mouse button; the middle mouse button is very inconvenient. Thanks!

### ISSUE THREAD

**Issue by wangenius** [excalidraw/excalidraw#8777]
I need to move the canvas with the right mouse button; the middle mouse button is very inconvenient. Thanks!

**Comment by bgevko** [comment]
There's another issue open for a similar feature request ([8753](https://github.com/excalidraw/excalidraw/issues/8753#issue-2631494257). I've been looking to contribute, is this something I can work on?

**Comment by Mrazator** [comment]
> When enabled, right-click will behave as a context menu on the right mouse up if the mouse did not move. Otherwise, it behaves as wheel / space pan event

Yeah, something like that might work. It could probably be available on both the wheel and right button without an additional item in the context menu (i.e. Miro seems to have this), but it has to work well.

> Can I work on this?

Sure, though I would propose trying to analyze what is the general consensus for panning across tools (middle + right button; configuration to switch middle / right button, something else, etc.). Also dig a bit into history why right button was not implemented -  there might be a reason why we didn't introduce it in the first place.

**Comment by jameswilliamknight** [comment]
I would appreciate a feature like this, even experimental. I use it in Obsidian, and it's uncomfortable switching between the Canvas feature and Excalidraw. Not a deal breaker!

**Comment by wangenius** [comment]
thx

**Comment by einarpersson** [comment]
My colleague has a similar issue with the mouse wheel being the mouse button to pan. He says however that he is used to using the side button on his mouse to pan in Figma/Figjam (and that it just works out-of-the-box somehow). He didn't bother to leave a comment here so I thought I'd do it for him...

So - perhaps check with Figma/Figjam/Penpot/Miro/Mural etc?
